src/server/Server.java

```diff
@@ -9,92 +15,72 @@
 import java.net.Socket;
 import java.nio.file.Path;
 import java.security.KeyStore;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
-public abstract class Server {
-    private final SparkDB MIME = new SparkDB();
-    private final HashMap<String, String> CustomHeaders = new HashMap<>();
-    private int Port = 80;
-    private int MaxConcurrentRequests = 1000;
-    private boolean GZip = true;
-    private int MaxRequestSizeKB = 100_000;
-    private int backlog = MaxConcurrentRequests * 5;
-    private String MIMEFile = "etc/MIME.db";
-    private String WWWDir = "www";
-    private int SOTimeout = 60_000;
-    private int BufferSize = 524288;
+public class Server {
+    public static final String HTTP_PROTO_VERSION = "HTTP/1.1";
 
-    private static String getStackTrace(final Throwable throwable) {
-        final StringWriter sw = new StringWriter();
-        final PrintWriter pw = new PrintWriter(sw, true);
-        throwable.printStackTrace(pw);
-        return sw.getBuffer().toString();
-    }
-
-    public void setSOTimeout(int in) {
-        SOTimeout = in;
-    }
-
-    public void setBufferSize(int in) {
-        BufferSize = in;
-    }
-
-    public void setMaximumConcurrentRequests(int in) {
-        MaxConcurrentRequests = in;
-        backlog = MaxConcurrentRequests * 5;
-    }
+    public final int port;
+    public final int maxRequestSizeBytes;
+    public final int maxConcurrentRequests;
+    public final int backlog;
+    public final boolean useGzip;
+    public final String rootDir;
+    public final String mimeFile;
 
-    public void setGZip(boolean in) {
-        GZip = in;
-    }
+    private final SparkDB MIME = new SparkDB();
+    private final Function<HttpRequest, HttpResponse> handler;
 
-    public void setMaximumRequestSizeInKB(int in) {
-        MaxRequestSizeKB = in;
-    }
 
-    public void setBacklog(int in) {
-        backlog = in;
+    public Server(Function<HttpRequest, HttpResponse> handler) {
+        this(8080, false, handler);
```

Comment:
port 80

Reply:
Requires root access while developing and anyways the user can change it when creating the server, it's just the default port.

Reply:
That should be noted in the README file